internal/utils/parse.go

```diff
@@ -0,0 +1,43 @@
+/*
+Apache Score
+Copyright 2022 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+*/
+package utils
+
+import (
+	"strconv"
+	"strings"
+)
+
+// TryParseJsonValue attempts to convert an input string into a simple JSON value (string, number, boolean, or null).
+//
+// Complex values (arrays and objects) are not supported and treated as strings.
+// Quoted values are always treated as strings.
+//
+// Conversion rules:
+//
+//	null    -> nil
+//	123     -> float64
+//	"123"   -> string
+//	false   -> boolean
+//	"false" -> string
+//	abc     -> string
+//	"abc"   -> string
+func TryParseJsonValue(str string) interface{} {
```

Comment:
Wouldn't it be better to just use a YAML parser here? There are some other ways strings can be created. For example using `'`. This is handy to allow for shell quotes. You would also get the ability to add objects etc which could also be useful.
We are outputting a YAML file so YAML input makes sense. Also: YAML is a superset of JSON, so JSON objects work as YAML objects too.

Parse float is also probably not sufficient as YAML supports man ways of expressing numbers: https://yaml.org/spec/1.2.2/#24-tags

Also this does not check for invalid strings:
- `"hello`
Also would break some valid strings:
- `"this string has a trailing escaped quote \""

Reply:
Switched to `yaml.Decode(..)`